### PR TITLE
Fix broken pause state when calling setMusic on playing audio element

### PR DESCRIPTION
**Pull request by ericdrobinson**
Changes introduced in PR #174 created a situation whereby the player UI could show a "pause" button even though the audio element was actually paused. This was due to:

1) The "pause" event being interrupted when setting `audio.src`.
2) The `Toggle()` function ignoring the state of the audio element, and instead being [bound to the play/pause events](https://github.com/MoePlayer/APlayer/commit/0b0be17c70ac93de51e5599b603486df3f8fa104).

Because the internal state _does_ change from "playing" to "paused", but the event never properly triggers, the `Toggle()` function breaks. This PR addresses the issue by duplicating the "pause" event handling code into the "abort" event, which _is_ called when the "pause" event is interrupted.

A longer description from the commit:
>It appears that the "pause" event is never triggered if you call `audio.pause()` on a playing audio element and then immediately set `audio.src`. The "abort" event, however, _is_ called in this case.
>
>Because the "pause" event isn't raised, the APlayer code that resets the pause button state doesn't trigger. This leaves the APlayer in a state where the pause button is still showing but the audio element _isn't_ actually playing. Because `APlayer.toggle()` (which is called by the Play/Pause button) does not check the state of the audio element, and instead checks the state of the UI, we end up in a situation where the pause button is visible but we're not playing. The call to Toggle, then, calls `APlayer.pause()` which checks to see if it's already paused or not. Because the audio element is already paused, the call is ignored and the event never triggers (which would reset the button UI).
>
>This [temporary] "fix" resolves the issue by intercepting the "abort" event that is triggered when the "pause" event is aborted. It then runs the same logic as the "pause" event, checking to see if the playback button is in a "paused" state and, if so, resetting it to a "play" button state.
>
>_In theory_, the checks at the beginning of the `pauseHandler` arrow function should be enough to safeguard any other unrelated calls to "abort".

## [FUTURE] Fixing This For Good
The **_correct_** way to handle this issue would be to adjust the APlayer code to properly handle the Promises returned by [audio.play()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play) and [audio.pause()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause). I did not implement a full solution because this would imply a potential change to [supported platforms](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play#Browser_compatibility). It appears that desktop browsers only recently started [returning Promises](https://developers.google.com/web/updates/2017/06/play-request-was-interrupted) and it is unclear what their support looks like on mobile platforms. As a result, the safest option would be to include a Promise-based approach as a default with something similar to the current implementation as a backup.

Unfortunately, such a fix is beyond the scope of this PR as it affects the entire player.

### CODE REVIEW

**Comment by ericdrobinson** [comment]
This PR incorporates (is built on) changes for PR #173 and #174.

File: src/APlayer.js [hunk 1]
```diff
@@ -857,6 +867,8 @@ class APlayer {
      * @param {Array} newMusic
      */
     addMusic(newMusic) {
+        let wasSingle = !this.isMultiple();
```
Comment (micooz) [inline src/APlayer.js]: Seems that `wasSingle` is not re-assigned below, please use `const` instead.

File: src/APlayer.js [hunk 2]
```diff
@@ -520,7 +525,7 @@ class APlayer {
                 }
             });
 
-            this.audio.addEventListener('pause', () => {
+            let pauseHandler = () => {
```
Comment (micooz) [inline src/APlayer.js]: Seems that `pauseHandler` is not re-assigned below, please use `const` instead.

Comment (ericdrobinson) [reply src/APlayer.js]: Handled with commit 1a4ea7a4ed4cacfc9e6686495e1e5b72a8e0e1cd.

File: src/APlayer.js (hunk 1 above)
Comment (ericdrobinson) [reply src/APlayer.js]: Handled with commit 1a4ea7a4ed4cacfc9e6686495e1e5b72a8e0e1cd.

**Pull request merged**